tests/test_upgrade_template.py

```diff
@@ -0,0 +1,106 @@
+"""
+Tests for voting ??/05/2023
```

Comment:
The testsuite should be checking that all "admin functionality" works as intended: can't finish if hadn't started, timestamps checks, view functions which revert under some conditions, etc.